### PR TITLE
Fix broken device option

### PR DESCRIPTION
**Pull request by Fatme**
Currently the same code was used for printing device information and for validating --device option.

### CODE REVIEW

**Comment by rosen-vladimirov** [comment]
:+1:

File: lib/iphone-simulator-xcode-6.js [hunk 1]
```diff
@@ -23,6 +38,21 @@ var XCode6Simulator = (function () {
...
 
+    XCode6Simulator.prototype.getDevicesInfo = function () {
+        var availableDeviceIdentifiers = this.getAvailableDevices();
+        var keys = _.keys(availableDeviceIdentifiers);
+        var result = [];
+
+        _.each(keys, function (deviceIdentifier) {
+            var devices = availableDeviceIdentifiers[deviceIdentifier];
+            _.each(devices, function (device) {
+                return result.push(device);
```
Comment (teobugslayer) [inline lib/iphone-simulator-xcode-6.js]: each + push -> map

File: lib/iphone-simulator-xcode-6.js [hunk 2]
```diff
@@ -23,6 +38,21 @@ var XCode6Simulator = (function () {
...
+            });
+        });
+
+        return result;
+    };
```
Comment (teobugslayer) [inline lib/iphone-simulator-xcode-6.js]: I think a more compact code would be
```
return _(this.getAvailableDevices())
	.map(function(availabledeIds, viceIdentifier) {
		return availableIds;
	})
	.flatten();
```

File: lib/iphone-simulator-xcode-6.ts [hunk 3]
```diff
@@ -50,11 +34,24 @@ export class XCode6Simulator implements ISimulator {
...
+		var keys = _.keys(availableDeviceIdentifiers);
+		var result: IDevice[] = [];
+
+		_.each(keys, deviceIdentifier => {
+			var devices = availableDeviceIdentifiers[deviceIdentifier];
+			_.each(devices, device => result.push(device));
+		});
+
+		return result;
+	}
```
Comment (teobugslayer) [inline lib/iphone-simulator-xcode-6.ts]: Again, as the code above?

```
return _(this.getAvailableDevices())
    .map(_.identity)
    .flatten()
    .value()
```

**Comment by teobugslayer** [comment]
:+1:

**Pull request merged**